packages/functional-tests/pages/login.ts

```diff
@@ -352,7 +352,7 @@ export class LoginPage extends BaseLayout {
 
   async submit(waitForNav = true) {
     if (waitForNav) {
```

Comment:
I think you should be able to remove anything `waitForNav` and possibly even `waitForNavOnSubmit` after the change, I'd added that as a conditional since `waitForNavigation` was causing problems there. Btw, why go with `waitForEvent('framenavigated')` over `waitForURL` in most of these, is it so we don't have to specify what URL it should be navigating to?

<img src="https://github.com/mozilla/fxa/assets/13018240/e4fc5965-4c6a-4970-a6e1-69542daa2bcf" width="450">

Reply:
> Yes, I found in the playwright docs that the next best thing to do to replace WaitForNavigation() is to use either waitForUrl() (in places where you have specified urls) or waitForEvent('framenavigated') (where there is multiple redirects as well as navigations). So I used both depending on situations.

👍 SGTM. I referenced this [here](https://github.com/mozilla/fxa/pull/15524/files#r1262968084) BTW.

> Also, I feel it would be still good to keep `waitForNav` and `waitForNavOnSubmit` as these are simply used as parameters as I have replaced the `waitForNavigation()` altogether everywhere.

I introduced those parameters because `waitForNavigation` was seemingly causing issues and I needed the option to skip that check for a couple of tests. With this update to `waitForEvent`, I don't think we should ever need the option to skip them. If it causes test failures, there might be something wonky with the test. I'm OK with leaving it though if you like having the option for it.